utilities/Microsoft.Quantum.Katas/CheckKataMagic.cs

```diff
@@ -162,21 +172,28 @@ public virtual bool Simulate(OperationInfo test, string userAnswer, IChannel cha
 
             try
             {
-                var qsim = CreateSimulator();
+                List<SimulatorBase> simulators = CreateSimulators(channel, test);
                 var hasWarnings = false;
 
-                qsim.DisableLogToConsole();
-                qsim.Register(skeletonAnswer.RoslynType, referenceAnswer.RoslynType, typeof(ICallable));
-                qsim.OnLog += (msg) =>
+                foreach(SimulatorBase qsim in simulators)
                 {
-                    hasWarnings = msg?.StartsWith("[WARNING]") ?? hasWarnings;
-                    channel.Stdout(msg);
-                };
+                    Logger.LogDebug($"Simulating test {test.FullName} on {qsim.GetType().Name}");
+                    var simHasWarnings = false;
 
-                var value = test.RunAsync(qsim, null).Result;
+                    qsim.DisableLogToConsole();
+                    qsim.Register(skeletonAnswer.RoslynType, referenceAnswer.RoslynType, typeof(ICallable));
 
-                if (qsim is IDisposable dis) { dis.Dispose(); }
+                    qsim.OnLog += (msg) =>
+                    {
+                        simHasWarnings = msg?.StartsWith("[WARNING]") ?? simHasWarnings;
+                        if(simHasWarnings == true) { hasWarnings = true; }
+                        channel.Stdout(msg);
```

Comment:
@tcNickolas @anjbur , it seems that code here check for any warnings in the reference implementation like Deprecated syntax. Is this needed to ensure that only the latest syntax is being used? Or could I possibly remove the code here and just do a `sim.OnLog += channel.Stdout;` .